src/averageAges.js

```diff
@@ -15,11 +15,19 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const isNeedMen = century
+    ? people.filter(person =>
+      person.sex === 'm'
+    && Math.ceil(person.died / 100) === century
+    )
+    : people.filter(person => person.sex === 'm');
+
+  const menLifeLength = isNeedMen.map(person => person.died - person.born);
+
+  const manAvarageAge = menLifeLength.reduce((sum, age) =>
+    sum + age) / menLifeLength.length;
```

Comment:
you use same functions several times below in the code, create helper functions for that